packages/plugin-survey/src/index.ts

```diff
@@ -165,165 +263,411 @@ class SurveyPlugin implements JsPsychPlugin<Info> {
-    const setup_likert_question = () => {};
-    const setup_multichoice_question = (question, question_params) => {
-      // required: options
-      // optional: columns, option_reorder
-      question.title = question_params.prompt;
-      question.isRequired = question_params.required;
-      question.choices = question_params.options;
-      if (typeof question_params.option_reorder == "undefined") {
-        question.choicesOrder = info.parameters.pages.nested.option_reorder.default; // TO DO: is this how to get the default value from parameter info?
-      } else {
-        question.choicesOrder = question_params.option_reorder;
-      }
-      question.colCount = question_params.columns;
-      return question;
-    };
-
-    const setup_ranking_question = () => {};
-    const setup_rating_question = () => {};
-    const setup_text_question = (question, question_params) => {
-      // optional: placeholder, textbox_rows, textbox_columns
-      question.title = question_params.prompt;
-      question.isRequired = question_params.required;
-      question.placeHolder = question_params.placeholder;
-      if (question.type == "comment") {
-        question.rows = question_params.textbox_rows;
-        question.cols = question_params.textbox_columns;
-      } else {
-        question.size = question_params.textbox_columns;
-      }
-      return question;
-    };
-
-    // https://surveyjs.io/Documentation/Library#survey-objects
-    // set up in code
-    const survey = new Survey();
-
     // navigation buttons
-    survey.pagePrevText = trial.button_label_back;
-    survey.pageNextText = trial.button_label_next;
-    survey.completeText = trial.button_label_finish;
+    this.survey.pagePrevText = this.params.button_label_back;
+    this.survey.pageNextText = this.params.button_label_next;
+    this.survey.completeText = this.params.button_label_finish;
 
     // page numbers
-    survey.showQuestionNumbers = trial.show_question_numbers;
+    this.survey.showQuestionNumbers = this.params.show_question_numbers;
 
     // survey title
-    if (trial.title !== null) {
-      survey.title = trial.title;
+    if (this.params.title !== null) {
+      this.survey.title = this.params.title;
     }
 
+    // required question label
+    this.survey.requiredText = this.params.required_question_label;
+
+    // TO DO: response validation
+    this.survey.checkErrorsMode = "onNextPage"; // onValueChanged
+
+    // TO DO: automatic response accuracy scoring for questions with a correctAnswer value
+    // see Survey Model onIsAnswerCorrect event
+    // https://surveyjs.io/Documentation/Library/?id=SurveyModel#onIsAnswerCorrect
+
     // pages and questions
-    for (let i = 0; i < trial.pages.length; i++) {
+    for (let i = 0; i < this.params.pages.length; i++) {
       let page_id = "page" + i.toString();
-      let page = survey.addNewPage(page_id);
-      if (trial.randomize_question_order) {
+      let page = this.survey.addNewPage(page_id);
```

Comment:
There are some places here were `const` would help to make the intention clear, such as here
```suggestion
      const page = this.survey.addNewPage(page_id);
```